gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2562,18 +2903,33 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
 
     if( !m_osGridMapping.empty() && nMainDimId == m_nRecordDimID )
     {
-        status =
-            nc_put_att_text(m_nLayerCDFId, nVarID, CF_GRD_MAPPING,
+        if ( !m_bLegacyCreateMode )
+        {
+            m_poDS->vcdf.nc_put_vatt_text(nVarID, CF_GRD_MAPPING, m_osGridMapping.c_str());
+        }
+
+        else
+        {
+            status =
+                nc_put_att_text(m_nLayerCDFId, nVarID, CF_GRD_MAPPING,
                             m_osGridMapping.size(), m_osGridMapping.c_str());
-        NCDF_ERR(status);
+            NCDF_ERR(status);
+        }
     }
 
     if( !m_osCoordinatesValue.empty() && nMainDimId == m_nRecordDimID )
     {
-        status = nc_put_att_text(m_nLayerCDFId, nVarID, CF_COORDINATES,
-                                 m_osCoordinatesValue.size(),
-                                 m_osCoordinatesValue.c_str());
-        NCDF_ERR(status);
+        if ( !m_bLegacyCreateMode )
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation